src/components/Users/UserListAndCard.tsx

```diff
@@ -81,9 +89,24 @@ const getNameAndStatusCard = (
       <div className="flex flex-row justify-between gap-x-3">
         <div className="flex flex-col">
           <div className="flex items-center gap-x-3">
-            <h1 id={`name-${user.username}`} className="text-base font-bold">
-              {formatName(user)}
-            </h1>
+            <TooltipProvider delayDuration={200}>
+              <Tooltip>
+                <TooltipTrigger asChild>
+                  <h1
+                    id={`name-${user.username}`}
+                    className="text-base font-bold truncate max-w-[7rem] inline-block"
+                  >
+                    {formatName(user)}
+                  </h1>
+                </TooltipTrigger>
+                <TooltipContent
+                  className="px-4 py-3 text-sm bg-black text-white rounded-xl shadow-xl border border-indigo-300 animate-in fade-in-0 zoom-in-95 data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2"
```

Comment:
Why add any of this? Shadcn already handles the animation, let's not modify that.

```suggestion
                  className="px-4 py-3 text-sm bg-black text-white rounded-md shadow-lg"
```